easybuild/easyblocks/generic/intelbase.py

```diff
@@ -253,35 +255,40 @@ def prepare_step(self, *args, **kwargs):
 
         # Decide if we need a license or not (default is True because of defaults of individual Booleans)
         self.requires_runtime_license = self.cfg['requires_runtime_license'] and requires_runtime_license
+        self.serial_number = self.cfg['serial_number']
 
-        if self.requires_runtime_license:
-            default_lic_env_var = 'INTEL_LICENSE_FILE'
-            license_specs = ensure_iterable_license_specs(self.cfg['license_file'])
-            lic_specs, self.license_env_var = find_flexlm_license(custom_env_vars=[default_lic_env_var],
-                                                                  lic_specs=license_specs)
-
-            if lic_specs:
-                if self.license_env_var is None:
-                    self.log.info("Using Intel license specifications from 'license_file': %s", lic_specs)
-                    self.license_env_var = default_lic_env_var
+        if self.serial_number:
+            self.log.info("Using provided serial number (%s) and ignoring other licenses", self.serial_number)
+        else:
+            if self.requires_runtime_license:
```

Comment:
@damianam this could be an `elif` (which also avoids all the extra indentation)?

```python
elif self.requires_runtime_license:
    ...